src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -391,6 +397,10 @@ void UDPEndPointImplLwIP::LwIPReceiveUDPMessage(void * arg, struct udp_pcb * pcb
     pktInfo->SrcPort     = port;
     pktInfo->DestPort    = pcb->local_port;
 
+    // Increase mDelayReleaseCount to delay release of this UDP EndPoint until on more HandleDataReceived is
+    // pending on this UDP EndPoint.
+    ep->mDelayReleaseCount++;
+
     CHIP_ERROR err = ep->GetSystemLayer().ScheduleLambda([ep, p = System::LwIPPacketBufferView::UnsafeGetLwIPpbuf(buf), pktInfo] {
         ep->HandleDataReceived(System::PacketBufferHandle::Adopt(p), pktInfo);
```

Comment:
```suggestion
        ep->mDelayReleaseCount--;
        ep->HandleDataReceived(System::PacketBufferHandle::Adopt(p), pktInfo);
```

I suggest decrease the count here rather than inside `HandleDataReceived`